src/component/CameraModal.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from "react";
+import { Camera } from "expo-camera";
+import { Modal, Text, TouchableOpacity, View } from "react-native";
+import { FontAwesome5 } from "@expo/vector-icons";
+
+const CameraModal = ({visible,setShowCamera,setFormData,formData}) => {
+  const [hasCameraPermission, setHasCameraPermission] = useState(null);
+  const [camera, setCamera] = useState(null);
+  const [type, setType] = useState(Camera.Constants.Type.back);
+
+  const takePicture = async () => {
+    if (camera) {
+      const data = await camera.takePictureAsync(null);
+      setFormData({...formData,picture:data.uri});
```

Comment:
formdata and setform data should not be done by this component 
camera modal should only handle taking picture and giving back the picture to its parent component @amitark7 
also code no t formatted